bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/sequencefiles/SequenceFileSink.java

```diff
@@ -0,0 +1,233 @@
+     * directory from the base output filename.
+     *
+     * <p>Default is a uniquely named sibling of baseOutputFilename, e.g. if baseOutputFilename is
+     * /path/to/foo, the temporary directory will be /path/to/temp-beam-foo-$date.
+     *
+     * @param sink The {@link SequenceFileSink} that will be used to configure this write operation.
+     * @param keyClass The class of the {@link SequenceFile} key.
+     * @param valueClass The class of the {@link SequenceFile} value.
+     * @param serializationNames A list of {@link Serialization} class names.
+     */
+    SeqFileWriteOperation(SequenceFileSink<K, V> sink, Class<K> keyClass,
+        Class<V> valueClass, String[] serializationNames) {
+      super(sink);
+      this.keyClass = keyClass;
+      this.valueClass = valueClass;
+      this.serializationNames = serializationNames;
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Writer<KV<K, V>> createWriter() throws Exception {
+      return new SeqFileWriter<>(this);
+    }
+  }
+
+  /**
+   * Wrapper for {@link SequenceFile.Writer} that adapts hadoop's {@link SequenceFile} api to Beam's
+   * {@link org.apache.beam.sdk.io.FileBasedSink.Writer} api.
+   *
+   * @param <K> The type of the {@link SequenceFile} key.
+   * @param <V> The type of the {@link SequenceFile} value.
+   */
+  private static class SeqFileWriter<K,V> extends Writer<KV<K,V>> {
+    private final SeqFileWriteOperation<K,V> writeOperation;
+    private SequenceFile.Writer sequenceFile;
+    private final AtomicLong counter = new AtomicLong();
+
+    /**
+     * Constructs a new {@link SeqFileWriter}.
+     *
+     * @param writeOperation The parent operation.
+     */
+    SeqFileWriter(SeqFileWriteOperation<K, V> writeOperation) {
+      super(writeOperation, MimeTypes.BINARY);
+      this.writeOperation = writeOperation;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void prepareWrite(WritableByteChannel channel) throws Exception {
+      LOG.info("Opening new writer");
+
+      Configuration configuration = new Configuration(false);
+      configuration.setStrings("io.serializations", writeOperation.serializationNames);
+
+      FSDataOutputStream outputStream = new FSDataOutputStream(new OutputStreamWrapper(channel), new Statistics("dataflow"));
+      sequenceFile = SequenceFile.createWriter(configuration,
+          SequenceFile.Writer.stream(outputStream),
+          SequenceFile.Writer.keyClass(writeOperation.keyClass),
+          SequenceFile.Writer.valueClass(writeOperation.valueClass),
+          SequenceFile.Writer.compression(SequenceFile.CompressionType.BLOCK)
+      );
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void finishWrite() throws Exception {
+      sequenceFile.hflush();
+      sequenceFile.close();
+      super.finishWrite();
+      LOG.info("Closing writer with " + counter.get() + " items");
```

Comment:
Would this info log be spammy?  Is this more of a debug thing?

Reply:
it would be printed a couple of hundred times (once per file).
You right the user doesn't really care